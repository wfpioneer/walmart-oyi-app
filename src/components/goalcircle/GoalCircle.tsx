import React, {useState, useEffect} from "react";
import styles from "./GoalCircle.style";
import { View, Text } from "react-native";

export interface Props {
    goalTitle: string,
    completionPercentage: number,
    completionGoal?: number
};

function GoalCircle(Props:Props) {
    let target;
    Props.completionGoal?(target=Props.completionGoal):(target=95);
    let atGoal=(Props.completionPercentage>=target)?styles.goalMet:styles.goalNotMet;
    let ringTwo=(Props.completionPercentage>=50)?atGoal:styles.under50;
    let transfOne=(Props.completionPercentage>=50)?180:(Props.completionPercentage*3.6);
    let transfTwo=(Props.completionPercentage>=50)?(Props.completionPercentage*3.6):0;
    return (
        <View>
            <View style={styles.baseRing}>
                <View style={[styles.container, {transform: [{rotate:`${transfOne}deg`}]}]}>
                    <View style={[atGoal, styles.halfRing]}/>
                </View>
                <View style={[styles.container, {transform: [{rotate: `${transfTwo}deg`}]}]}>
                    <View style={[ringTwo, styles.halfRing]}/>
                </View>
                <View style={styles.centerRing}>
                    <Text style={styles.goalDisp}>{Props.completionPercentage}%</Text>
                </View>
            </View>
            <Text style={styles.freq}>Daily</Text>
            <Text style={styles.goalNameActive}>Items</Text>
        </View>
    )
};

export default GoalCircle;