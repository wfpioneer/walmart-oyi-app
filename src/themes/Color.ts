export const COLOR = {
  // Ex: BLUE_GREY_900, YELLOW_PRIMARY, GREY_300
  MAIN_THEME_COLOR: '#0071e9',
  BLACK_TRANSPARENT_200: '#00000033',
  BLACK_TRANSPARENT_500: '#00000080',
  BLACK_TRANSPARENT_600: '#00000099',

  BLACK: '#000000',
  GRAY: '#888888',
  TRANSPARENT: '#00000000',

  GREY_50: '#fafafa',
  GREY_100: '#f5f5f5',
  GREY_200: '#eeeeee',
  GREY_300: '#e0e0e0',
  GREY_400: '#bdbdbd',
  GREY_500: '#9e9e9e',
  GREY_600: '#757575',
  GREY_700: '#616161',
  GREY_800: '#424242',
  GREY_900: '#212121',

  RED_50: '#ffebee',
  RED_100: '#ffcdd2',
  RED_150: '#b42d02',
  RED_200: '#ef9a9a',
  RED_300: '#e57373',
  RED_400: '#ef5350',
  RED_500: '#f44336',
  RED_550: '#ab2703',
  RED_600: '#e53935',
  RED_700: '#d32f2f',
  RED_800: '#c62828',
  RED_900: '#b71c1c',

  TRACKER_RED: '#D6512D',
  TRACKER_GREY: '#545F7A',

  HOLO_GREEN_DARK: '#669900',
  LIGHT_AQUA_BLUE: '#F0F9FF',
  TIP_BLUE: '#1977d3',

  // Native Color

  SECONDARY_NEGATIVE_COLOR: '#ec0439',

  WHITE: '#ffffffff',
  RED: '#ec0439',
  GREEN: '#33691E',

  GREY: '#c5c5c5',
  ORANGE: '#e74517',
  MID_GREY: '#eaeaea',
  LIGHT_GRAY: '#f6f6f6',
  PALE_GREEN: '#EAFCDA',
  DISABLED_BLUE: '#5FA7F2',
  TRAINING_BLUE_DARK: '#0039a9',
  TRAINING_BLUE: '#006DE5',
  TRAINING_BLUE_DISABLED: '#9FA8DA',
  MODERATE_RED: '#ffe5dd',
  MINE_SHAFT: '#3c3c3c',
  SHIP_COVE: '#8492a8',
  GREY_TRANSPARENT: '#f4f4f9',
  MODERATE_BLACK: '#545f7a',
  GAINSBORO: '#d8d8d8',
  MIDNIGHT: '#323a4e',
  PALE_BLUE: '#dbecff',
  PALE_GRAY: '#edededcc',
  PINK: '#c71585'
};

export default COLOR;
