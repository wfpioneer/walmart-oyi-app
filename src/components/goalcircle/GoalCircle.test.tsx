import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import GoalCircle from './GoalCircle';
import { COLOR } from '../../themes/Color';

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'mockMaterialCommunityIcons');

describe('GoalCircle Component', () => {
  it('Renders an GoalCircle  with goal tiltle, active as false and completion perc 40', () => {
    const { toJSON } = render(
      <GoalCircle
        goalTitle="Test"
        completionGoal={100}
        completionPercentage={40}
        active={false}
        frequency="70"
        onPress={jest.fn()}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('Renders an GoalCircle with goal tiltle, active as false and completion perc 100', () => {
    const { toJSON, getByTestId } = render(
      <GoalCircle
        goalTitle="Test"
        completionGoal={100}
        completionPercentage={100}
        active={false}
        frequency="70"
        onPress={jest.fn()}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const textComp = getByTestId('txtGoalTitle');
    expect(textComp.props.style.backgroundColor).toBe(COLOR.GREY_300);
  });

  it('Renders an GoalCircle  with goal tiltle, active as true and completion perc 100', () => {
    const { getByTestId, toJSON } = render(
      <GoalCircle
        goalTitle="Test"
        completionGoal={100}
        completionPercentage={100}
        active={true}
        frequency="70"
        onPress={jest.fn()}
      />
    );
    expect(toJSON()).toMatchSnapshot();
    const textComp = getByTestId('txtGoalTitle');
    expect(textComp.props.style.backgroundColor).toBe(COLOR.TRAINING_BLUE);
  });

  it('should call mock function on click of goal title', () => {
    const mockGoalClick = jest.fn();
    const { getByTestId } = render(
      <GoalCircle
        goalTitle="Test"
        completionGoal={100}
        completionPercentage={100}
        active={true}
        frequency="70"
        onPress={mockGoalClick}
      />
    );
    const btnGoalTitleClick = getByTestId('btnGoalTitleClick');
    fireEvent.press(btnGoalTitleClick);
    expect(mockGoalClick).toBeCalledTimes(1);
  });
});
