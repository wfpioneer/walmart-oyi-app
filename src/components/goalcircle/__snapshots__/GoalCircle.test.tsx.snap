// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalCircle Component Renders an GoalCircle  with goal tiltle, active as false and completion perc 40 1`] = `
<View
  style={
    {
      "marginHorizontal": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#e0e0e0",
        "borderRadius": 40,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 80,
      }
    }
  >
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "144deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#D6512D",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "0deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#e0e0e0",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderRadius": 33,
          "height": 65,
          "justifyContent": "center",
          "width": 65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        40
        %
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="btnGoalTitleClick"
  >
    <Text
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderRadius": 30,
          "color": "#006DE5",
          "height": 30,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 100,
        }
      }
      testID="txtGoalTitle"
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`GoalCircle Component Renders an GoalCircle  with goal tiltle, active as true and completion perc 100 1`] = `
<View
  style={
    {
      "marginHorizontal": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#e0e0e0",
        "borderRadius": 40,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 80,
      }
    }
  >
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "180deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#33691E",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "360deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#33691E",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderRadius": 33,
          "height": 65,
          "justifyContent": "center",
          "width": 65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        100
        %
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="btnGoalTitleClick"
  >
    <Text
      style={
        {
          "backgroundColor": "#006DE5",
          "borderRadius": 30,
          "color": "#ffffffff",
          "height": 30,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 100,
        }
      }
      testID="txtGoalTitle"
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`GoalCircle Component Renders an GoalCircle with goal tiltle, active as false and completion perc 100 1`] = `
<View
  style={
    {
      "marginHorizontal": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#e0e0e0",
        "borderRadius": 40,
        "height": 80,
        "justifyContent": "center",
        "marginBottom": 10,
        "width": 80,
      }
    }
  >
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "180deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#33691E",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "height": 80,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 80,
          },
          {
            "transform": [
              {
                "rotate": "360deg",
              },
            ],
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#33691E",
            },
            {
              "borderBottomLeftRadius": 40,
              "borderTopLeftRadius": 40,
              "height": 80,
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 40,
            },
          ]
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffffff",
          "borderRadius": 33,
          "height": 65,
          "justifyContent": "center",
          "width": 65,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        100
        %
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="btnGoalTitleClick"
  >
    <Text
      style={
        {
          "backgroundColor": "#e0e0e0",
          "borderRadius": 30,
          "color": "#006DE5",
          "height": 30,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": 100,
        }
      }
      testID="txtGoalTitle"
    >
      Test
    </Text>
  </View>
</View>
`;
